hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1011,7 +1011,7 @@ public String startDeletePartitionCommit(HoodieTableMetaClient metaClient) {
    */
   @VisibleForTesting
   String startCommit(Option<String> providedInstantTime, String actionType, HoodieTableMetaClient metaClient) {
-    if (needsUpgrade(metaClient)) {
+    if (UpgradeDowngrade.needsUpgrade(metaClient, config, config.getWriteVersion())) {
```

Comment:
this seems unnecessary right to have another param for write version since we are already passing the config?